### PR TITLE
hie-core Logging and exceptions

### PR DESCRIPTION
**Pull request by neil-da**
Before we logged (to the server) a pretty random set of errors - things like the user not writing down enough exposed modules. Now we only only errors that are a serious failing on our part, and in particular if there is an exception, we log that. I also tidied up some of the logging code in da-hs-base, but it still needs a flamethrower taking to it at some point in the future.

### CODE REVIEW

File: daml-foundations/daml-ghc/daml-compiler/src/DA/Service/Daml/Compiler/Impl/Handle.hs [hunk 1]
```diff
@@ -176,7 +175,7 @@ buildDar service file mbExposedModules pkgName sdkVersion buildDataFiles dalfInp
       let pkgModuleNames = S.fromList $ map T.unpack $ LF.packageModuleNames pkg
       let missingExposed = S.fromList (fromMaybe [] mbExposedModules) S.\\ pkgModuleNames
       unless (S.null missingExposed) $ do
-          liftIO $ IdeLogger.logSeriousError (ideLogger service) $
+          liftIO $ IdeLogger.logInfo (ideLogger service) $
```
Comment (cocreature) [inline daml-foundations/daml-ghc/daml-compiler/src/DA/Service/Daml/Compiler/Impl/Handle.hs]: Unless I am missing something this means that the error will not be shown by default. The right solution here is probably to not go via the logger at all and instead emit an error diagnostic.

Comment (neil-da) [reply daml-foundations/daml-ghc/daml-compiler/src/DA/Service/Daml/Compiler/Impl/Handle.hs]: I don't think this was ever right - call logError and then `fail` isn't the way to go. I have fixed it for now by just turning it into a `fail` with an error message, which should be no worse than before, but the whole thing is pretty dodgy. I added a FIXME.

Comment (neil-da) [reply daml-foundations/daml-ghc/daml-compiler/src/DA/Service/Daml/Compiler/Impl/Handle.hs]: I don't think this was ever right - call logError and then `fail` isn't the way to go. I have fixed it for now by just turning it into a `error` with an error message, and abort the whole thing, which should be no worse than before, but the whole thing is pretty dodgy. I added a FIXME.

Comment (neil-da) [reply daml-foundations/daml-ghc/daml-compiler/src/DA/Service/Daml/Compiler/Impl/Handle.hs]: @cocreature can you confirm this seems OK temporarily?

Comment (cocreature) [reply daml-foundations/daml-ghc/daml-compiler/src/DA/Service/Daml/Compiler/Impl/Handle.hs]: LGTM :+1:

**Pull request merged**